classes/class-swsales-metaboxes.php

```diff
@@ -707,6 +734,20 @@ public static function save_swsales_metaboxes( $post_id, $post ) {
 			update_post_meta( $post_id, 'swsales_sale_type', sanitize_text_field( $_POST['swsales_sale_type'] ) );
 		}
 
+		if ( ! empty( $_POST['swsales_hide_banner_by_role'] ) && is_array( $_POST['swsales_hide_banner_by_role'] )) {
+			$swsales_hide_banner_by_role = array_map( 'sanitize_text_field', $_POST['swsales_hide_banner_by_role'] );
+			update_post_meta( $post_id, 'swsales_hide_banner_by_role', wp_json_encode( $swsales_hide_banner_by_role ) );
```

Comment:
Note here that these wp_json_encode calls are not really needed as far as I know (assuming the wp_json_decode calls are also removed when the post meta is gotten), because the post meta get/set functions will serialize/deserialize the array anyway.

The serialize functions (automatically applied to arrays get/set in meta) are like old school json encode/decode.

But it seems like we were already saving these values this way, so let's continue to do that... better than coding some way to migrate old meta or account for both formats.

If I'm mistaken and the json encode is doing something important here, let us know. Update the comments. And share with the dev team.

Reply:
I asked ChatGPT and seems to think that meta isn't serialized automatically. IDK. This was how we had the level info stored so used the same model for the role info. I don't know if its needed but can test it out.